### PR TITLE
Work around C17s and Frigates not being able to deliver units

### LINKED ISSUE #16239: [TD] 2 in 1: a) Support Plane is not go away  b) Units stay over factory in one mapcell
a) Purple Player's Plane PWin
b) Red Player's units
See replay: https://cloud.mail.ru/public/37bE/LobyAByxd

### PR DESCRIPTION
**Pull request by abcdefg30**
Currently no units are delivered since the plane never lands.

### CODE REVIEW

**Comment by GraionDilach** [comment]
You can test the ticket with selling the airstrip after the production finished but before the plane landing.

**Comment by abcdefg30** [comment]
Then it appears to be already fixed on bleed.

File: mods/cnc/rules/aircraft.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ TRAN:
 		LandWhenIdle: true
 		TurnSpeed: 5
 		Speed: 150
-		LandableTerrainTypes: Clear,Rough,Road,Ore,Beach,Tiberium,BlueTiberium
+		LandableTerrainTypes: Clear,Rough,Road,Beach,Tiberium,BlueTiberium
```
Comment (pchote) [inline mods/cnc/rules/aircraft.yaml]: Can you please add spaces after the commas while you're here?

File: mods/cnc/rules/aircraft.yaml [hunk 2]
```diff
@@ -181,6 +181,7 @@ C17:
 		Speed: 326
 		Repulsable: False
 		MaximumPitch: 36
+		LandableTerrainTypes: Clear,Rough,Road,Beach
```
Comment (pchote) [inline mods/cnc/rules/aircraft.yaml]: And here.

**Review by pchote — CHANGES_REQUESTED** [review]
The D2k starport suffers from the same bug. Can you please fix that here too?

**Comment by abcdefg30** [comment]
Updated.

**Comment by pchote** [comment]
#16239 was filed based on playtest-20190209, well before the landing changes. Removed the reference from the PR description that would have automatically closed the issue.

**Comment by pchote** [comment]
This fix reveals a second regression: If a second plane arrives while the first is landing it will be blocked, and do a loop before landing. We can introduce an `OccupiesSpace` flag on `AircraftInfo` that disables the `ActorMap` influence adding/removing.

**Comment by pchote** [comment]
A third landing terrain type regression was noted in #16481.

> > idle yaks will not respond to queued orders (we had this issue before)
> 
> This issue is present in bleed as well and was regressed by #16365. the problem is that `LandWhenIdle` defaults to true even for aircraft that have no valid terrain for landing, so the aircraft gets stuck in a holding pattern until canceled. This should be handled in a separate fixup PR.

**Comment by tovl** [comment]
The last issue was already fixed in #16509 . Also, the work-arounds proposed here would heavily conflict with that PR. Expect a proper fix to be added soon to that PR or a follow up.

**Comment by abcdefg30** [comment]
> This fix reveals a second regression: [...]

This was not unexpected, as I said on IRC yesterday.

**Review by pchote — APPROVED** [review]
This bug is blocking meaningful testing of the AI, so we need a fix ASAP.
We can revert parts of this again in #16509 once that is ready.

**Review by ghost — APPROVED** [review]
Fixes the bug.

**Comment by reaperrr** [comment]
https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/22a1dc65c33c8b716c4a0d07e9b038e580914ca7...1c02a9a5777f5cd190aaf8adb68d96c196f6f124

**Pull request merged**